Kitodo/src/main/java/org/kitodo/production/renderer/KitodoMediaRenderer.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * (c) Kitodo. Key to digital objects e. V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo project.
+ *
+ * It is licensed under GNU General Public License version 3 or later.
+ *
+ * For the full copyright and license information, please read the
+ * GPL3-License.txt file that was distributed with this source code.
+ */
+
+package org.kitodo.production.renderer;
+
+import java.io.IOException;
+import java.util.Optional;
+
+import javax.faces.component.UIComponent;
+import javax.faces.component.UIParameter;
+import javax.faces.context.FacesContext;
+import javax.faces.context.ResponseWriter;
+
+import org.primefaces.component.media.Media;
+import org.primefaces.component.media.MediaRenderer;
+import org.primefaces.component.media.player.MediaPlayer;
+import org.primefaces.model.StreamedContent;
+import org.primefaces.util.AgentUtils;
+import org.primefaces.util.HTML;
+
+public class KitodoMediaRenderer extends MediaRenderer {
+
+    private static final String PLAYER_HTML_VIDEO = "html-video";
+
+    @Override
+    public void encodeEnd(FacesContext context, UIComponent component) throws IOException {
+        Media media = (Media) component;
+        ResponseWriter writer = context.getResponseWriter();
+        String src;
+        try {
+            src = getMediaSrc(context, media);
+        } catch (Exception ex) {
+            throw new IOException(ex);
+        }
+
+        Object value = media.getValue();
+        if (value instanceof StreamedContent) {
+            if (PLAYER_HTML_VIDEO.equals(media.getPlayer())) {
+                buildVideoTag(context, media, writer, src);
+                return;
+            }
+        }
+
+        boolean isIE = AgentUtils.isIE(context);
+        MediaPlayer player = resolvePlayer(context, media);
+        String sourceParam = player.getSourceParam();
+
+        // Fix npe exception https://forum.primefaces.org/viewtopic.php?t=62518
+        // This fix is deprecated cause primeface fix this in 7.0.13
```

Comment:
Does this mean we do not need the fix anymore because we recently updated to PrimeFaces 8?